snippets: matter: Added snippet for matter debug purposes (#19973)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/getting_started/advanced_kconfigs.rst

   :class: highlight

   west build -b nrf52840dk/nrf52840 -- -Dlock_SNIPPET=matter-diagnostic-logs

.. _ug_matter_debug_snippet:

Debug snippet
=============

> Matter debug snippet allows you to enable additional debug features while using Matter samples.

Reviewer: ```suggestion
The Matter debug snippet allows you to enable additional debug features while using Matter samples.
```

> Using this snippet there are the following features enabled:

Reviewer: ```suggestion
The following features are enabled when using this snippet:
```

>   * UART speed is increased to 1Mbit/s.

Reviewer: ```suggestion
  * UART speed is increased to 1 Mbit/s.
```

  * Log buffer size is set to high value to allow showing all logs.
  * Deferred mode of logging.
  * Increased verbosity of Matter logs.
  * Openthread is built from sources.
  * Openthread shell is enabled.
  * Openthread logging level is set to INFO.
  * Full shell functionalities.

>   * Logging source code location on VerifyOrDie failure which occurs in Matter stack.

Reviewer: ```suggestion
  * Logging source code location on VerifyOrDie failure that occurs in the Matter stack.
```

To use the snippet when building a sample, add ``-D<project_name>_SNIPPET=matter-debug`` to the west arguments list.

> Example for the ``nrf52840dk/nrf52840`` board target and the :ref:`matter_lock_sample` sample:

Reviewer: ```suggestion
For example, for the ``nrf52840dk/nrf52840`` board target and the :ref:`matter_lock_sample` sample, use the following command:
```
